sound/soc/sof/topology.c

```diff
@@ -1301,6 +1396,40 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 		return -EINVAL;
 	}
 
+	if (swidget->num_sink_pins > 1) {
+		ret = sof_alloc_pin_binding(swidget, true);
+		if (ret < 0) {
+			dev_err(scomp->dev, "failed to alloc sink binding array for %s\n",
+				w->name);
+			return ret;
+		}
+
+		ret = sof_parse_pin_binding(swidget, priv, true);
+		if (ret < 0) {
+			dev_err(scomp->dev, "failed to parse sink pin binding for %s\n",
+				w->name);
+			sof_free_pin_binding(swidget, true);
+			return ret;
+		}
+	}
```

Comment:
sorry I don't understand this code. If there is no special binding used for e.g. a mixer, why would we allocate memory that will not be used? It looks like you are conflating number of pins and special binding, those are different concepts, aren't they?

if I look at the next patch, you wrote "If widget has more than one sink/sink pins, and pin binding
array is not defined, Linux ID allocation will be used to
allocate queue ID dynamically."

So why would we always allocate memory for the bindings if they are not used? Something does not add up.